src/CqlBuilderPanel/definitionsSection/expressionSection/ExpressionEditor.tsx

```diff
@@ -135,6 +135,10 @@ export default function ExpressionEditor(props: ExpressionsProps) {
     predefinedFunctionsNames,
   ]);
 
+  useEffect(() => {
+    formik.setFieldValue("name", null);
+  }, [namesOptions]);
```

Comment:
we might be able to avoid useEffect just by calling onChange on namesOptions selectbox.
```
onChange={(evt) => {
    formik.setFieldValue("type", evt.target.value);
    formik.setFieldValue("name", "");
}}
```